src/uct/cuda/cuda_copy/cuda_copy_md.c

```diff
@@ -30,12 +30,14 @@ static ucs_config_field_t uct_cuda_copy_md_config_table[] = {
 
 static ucs_status_t uct_cuda_copy_md_query(uct_md_h md, uct_md_attr_t *md_attr)
 {
-    md_attr->cap.flags            = UCT_MD_FLAG_REG;
+    md_attr->cap.flags            = UCT_MD_FLAG_REG | UCT_MD_FLAG_ALLOC;
     md_attr->cap.reg_mem_types    = UCS_BIT(UCS_MEMORY_TYPE_HOST);
+    md_attr->cap.alloc_mem_types  = UCS_BIT(UCS_MEMORY_TYPE_CUDA) |
+                                    UCS_BIT(UCS_MEMORY_TYPE_CUDA_MANAGED);
     md_attr->cap.access_mem_type  = UCS_MEMORY_TYPE_CUDA;
     md_attr->cap.detect_mem_types = UCS_BIT(UCS_MEMORY_TYPE_CUDA) |
                                     UCS_BIT(UCS_MEMORY_TYPE_CUDA_MANAGED);
-    md_attr->cap.max_alloc        = 0;
+    md_attr->cap.max_alloc        = ULONG_MAX;
```

Comment:
That was what the original PR did but decided to remove that because of two reasons:
- `cuMemAllocManaged` can allocate as much as `malloc`
- max_alloc is not really checked anywhere and I think it's ok to just fail during uct_mem_malloc call if the device can't allocate beyond a certain size

Reply:
@bureddy Is the approach ok here?